Remove index shift by +-0.5 in voxel.py

wkentaro opened the pull request:
Close https://github.com/mikedh/trimesh/issues/370

**NOTE**: This is not backward compatible.

mikedh commented:
Nice! This was sort of haphazard before, it's great to make it consistent. A few quick thoughts:

- could we unit test?
- are there other places do we do the conversion points->indices? We should probably be consistent
- Maybe a quick blurb in the `voxel.py` top docstring which says something like "all voxels are referenced by the **center** of their cell box"

wkentaro commented:
> could we unit test?

I fixed the unit test just now.

> are there other places do we do the conversion points->indices? We should probably be consistent

As far as I know, the conversion is only used in `voxel.py` file. So I think now is good. 
The only concern is someone can assume `Voxel.as_boxes` returns boxes in the previous coordinate system. (origin is right-top of the first box)

> Maybe a quick blurb in the voxel.py top docstring which says something like "all voxels are referenced by the center of their cell box"

I agree. I added the docstring for Voxel class.

mikedh commented:
Awesome, thanks!
